docs/API/metadata.md

```diff
@@ -7,9 +7,30 @@ authenticated user, while other metadata is maintained internally by the server
 and can't be changed. Authenticated users can also add any additional metadata
 that might be of use.
 
->__NOTE__: right now the ability to search and filter using metadata is
-limited, but our intent is to be able to use any defined metadata value both
-for searches and to filter the results of [datasets/list](V1/list.md).
+Dataset metadata is represented as a set of nested JSON objects. There are four
+distinct key namespaces. These can be addressed (read or changed) at any level
+of the hierarchy using a dotted name path, for example `dataset.resource_id`
+for a dataset's resource ID, or `global.environment.cluster.ip`. The keys are
+lowercase alphabetic, plus digits, hyphen, and underscore: so `global.u-7` is
+OK, but `global.Frank` isn't.
+
+The four namespaces are:
+* `dataset` provides inherent attributes of the dataset, including the full
+`metadata.log` as `dataset.metalog`. Most of these attributes cannot be changed
+after creation.
```

Comment:
I don't think that would help, nor is that quite complete or accurate since the `metadata.log` key space is two levels deep. I wrestled with this very briefly; but, again, a full documentation of the agent/server agreed tarball format and `metadata.log` conventions would be more appropriate for details and that's (mostly) outside the scope of "API documentation". (Though not entirely, and maybe I ought to devote a cycle or two to thinking about adding it in this tree..)

Reply:
My principal concern was that the paragraph has a `blah.blah` next to a different `blah.blah`, and one is not a metadata key path....

But, nevermind.